collections/codec/indexing.go

```diff
@@ -77,7 +77,9 @@ func FallbackSchemaCodec[T any]() SchemaCodec[T] {
 				Kind: kind,
 			}},
 			// these can be nil because T maps directly to a schema value for this kind
-			ToSchemaType:   nil,
+			ToSchemaType: func(t T) (any, error) {
```

Comment:
Instead can we just have the `nil` case be handled properly?

Reply:
Done, it was already there for keys